webapp/channels/src/components/properties_card_view/propertyValueRenderer/user_property_renderer/userPropertyRenderer.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useRef} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import type {
+    PropertyField,
+    PropertyValue,
+} from '@mattermost/types/properties';
+
+import {getMissingProfilesByIds} from 'mattermost-redux/actions/users';
+import {getUser} from 'mattermost-redux/selectors/entities/users';
+
+import PreviewPostAvatar from 'components/post_view/post_message_preview/avatar/avatar';
+import UserProfileComponent from 'components/user_profile';
+
+import type {GlobalState} from 'types/store';
+
+import './user_property_renderer.scss';
+import {SelectableUserPropertyRenderer} from './selectable_user_property_renderer';
+
+type Props = {
+    field: PropertyField;
+    value: PropertyValue<unknown>;
+}
+
+export default function UserPropertyRenderer({field, value}: Props) {
+    const dispatch = useDispatch();
+    const loaded = useRef<boolean>(false);
+
+    const userId = value.value as string;
+    const user = useSelector((state: GlobalState) => getUser(state, userId));
+
+    useEffect(() => {
+        if (!loaded.current && userId && !user) {
+            dispatch(getMissingProfilesByIds([userId as string]));
+            loaded.current = true;
+        }
+    }, [dispatch, user, userId]);
```

Comment:
correct me if I am not wrong, but the loaded ref never gets reseted. I mean, If the userId change then it will not fetch the new user because loaded.current is still true?

Reply:
I've updated this to use the useUser hook, so all this logic is now not needed anymore.